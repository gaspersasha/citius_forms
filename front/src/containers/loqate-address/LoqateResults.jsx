import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { INPUT_STATUS } from '~constants';
import s from './styles/postcode.module.sass';

const LoqateResults = ({ lines, className, status }) => {
  const classes = cn(
    {
      // 'success-input-border': status === INPUT_STATUS.VALID,
      'error-input-border': status === INPUT_STATUS.INVALID,
    },
    ...className,
    s.border
  );

  return (
    <div className={classes} data-id="loqate-results">
      {lines.map(
        (line) =>
          Boolean(line && line.replace(/\s+/g, '').length) && (
            <p key={line}>{line.replace(/^\s+/g, '')}</p>
          )
      )}
    </div>
  );
};

LoqateResults.propTypes = {
  lines: PropTypes.arrayOf(PropTypes.string).isRequired,
  className: PropTypes.arrayOf(PropTypes.string),
  status: PropTypes.oneOf([
    INPUT_STATUS.DEFAULT,
    INPUT_STATUS.VALID,
    INPUT_STATUS.INVALID,
  ]),
};

LoqateResults.defaultProps = {
  className: [],
  status: INPUT_STATUS.DEFAULT,
};

export default LoqateResults;
