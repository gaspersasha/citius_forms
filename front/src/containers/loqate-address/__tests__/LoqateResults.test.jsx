import React from 'react';
import LoqateResults from '../LoqateResults';

describe('<LoqateResults />', () => {
  it('renders without errors', () => {
    const wrapper = shallow(<LoqateResults lines={['test0', 'test1']} />);

    expect(wrapper).toMatchSnapshot();
  });
});
