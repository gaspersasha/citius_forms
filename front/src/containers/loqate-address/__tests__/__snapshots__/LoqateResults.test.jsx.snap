// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoqateResults /> renders without errors 1`] = `
<div
  className="border"
  data-id="loqate-results"
>
  <p
    key="test0"
  >
    test0
  </p>
  <p
    key="test1"
  >
    test1
  </p>
</div>
`;
